### ISSUE TITLE
Visualization doesn't compile with latest CogServer

### ISSUE THREAD

**Issue by vsbogd** [opencog/visualization#1]
```
The following components WILL NOT be built:
-----------------------------------------------
   Doxygen         - Code documentation.
   GTK Visualizer  - GTK3-based Atomspace Visualizer.

-- Configuring done
-- Generating done
-- Build files have been written to: /ws/visualization/build
Scanning dependencies of target dotty
[ 25%] Building CXX object opencog/dotty/CMakeFiles/dotty.dir/DottyModule.cc.o
/ws/visualization/opencog/dotty/DottyModule.cc: In member function ‘std::__cxx11::string opencog::DottyModule::do_dotty(opencog::Request*, std::__cxx11::list<std::__cxx11::basic_string<char> >)’:
/ws/visualization/opencog/dotty/DottyModule.cc:162:49: error: cannot call member function ‘opencog::AtomSpace& opencog::BaseServer::getAtomSpace()’ without object
     AtomSpace *space = &CogServer::getAtomSpace();
                                                 ^
opencog/dotty/CMakeFiles/dotty.dir/build.make:82: recipe for target 'opencog/dotty/CMakeFiles/dotty.dir/DottyModule.cc.o' failed
make[2]: *** [opencog/dotty/CMakeFiles/dotty.dir/DottyModule.cc.o] Error 1
CMakeFiles/Makefile2:201: recipe for target 'opencog/dotty/CMakeFiles/dotty.dir/all' failed
make[1]: *** [opencog/dotty/CMakeFiles/dotty.dir/all] Error 2
Makefile:171: recipe for target 'all' failed
make: *** [all] Error 2

Exited with code exit status 2
```

**Comment by linas** [comment]
BaseServer was removed 5 or 7 years ago ....

**Comment by linas** [comment]
or maybe 7 years ago...

**Comment by vsbogd** [comment]
Do you mean this repo is deprecated and should be archived?

**Comment by linas** [comment]
I won't called it "deprecated", but it is unmaintained and maybe should be archived.

**Comment by linas** [comment]
OK, below is a better answer. but first a disclaimer: I wrote none of the code, and never used any of it. The history is lost to git: the files were originally in other directories, and file-moves cause that history to be lost.  (well, the opencog repo still has that history -- that's why the opencog repo has this huge number of contributors)

This repo contains three different visualization tools:
* a dotty exporter
* a GTK app
* a Tulip exporter.

I assume they all worked, once upon a time.  The GTK visualizer is probably very under-powered, and so not useful. The dotty and tulip exporters are probably incomplete.   Exporters are not hard to write -- I cobbled together a "graph markup language" exporter in a few hours or less, used it for the graph generation work. The biggest problem is that the visualization tools are  ... slow .. very slow ... and have a big learning curve to learn how to use properly (I used Gephi, it took me several hours to figure out how to use it at the "novice" level .. I couldn't figure out how to use CytoScape at all) 

So sure, marking this as "archived" might be OK, since it seems very unlikely that anyone will ever become interested in  this code...
